lib/filter-stream.js

```diff
@@ -176,7 +177,7 @@ function FilterStream( blockMap, options ) {
     var buffer = Buffer.concat( this._buffers )
     var offset = 0
 
-    while( offset < buffer.length ) {
+    while( buffer.length - offset >= this.blockSize ) {
```

Comment:
> Are we ignoring the last 4 bytes of buffer?

Nope, that part's just hidden in this diff; see lines 185-188 – the rest is stashed for the next transform (lines 170-175)